runtime/include/QuantumDevice.hpp

```diff
@@ -55,310 +57,558 @@ struct QuantumDevice {
+    //  EXECUTION MANAGEMENT
+    // ----------------------------------------
 
     /**
-     * @brief Set the number of device shots.
+     * @brief Set the number of execution shots.
+     *
+     * The Runtime will call this function to set the number of times the quantum execution is to
+     * be repeated. Generally, it will be called once before any quantum instructions are run, but
+     * some devices may choose to support it at arbitrary points in the program if they have the
+     * capability to simulate shot noise.
      *
-     * @param shots The number of noise shots
+     * Devices with no or restricted support are encouraged to raise a runtime error.
+     *
+     * @param shots Shot number.
      */
     virtual void SetDeviceShots(size_t shots) = 0;
 
     /**
-     * @brief Get the number of device shots.
+     * @brief Get the number of execution shots.
      *
-     * @return `size_t`
+     * @return `size_t` Shot number.
      */
-    [[nodiscard]] virtual auto GetDeviceShots() const -> size_t = 0;
+    virtual auto GetDeviceShots() const -> size_t = 0;
 
     /**
-     * @brief Set the PRNG of the device.
+     * @brief (optional) Set the PRNG of the device.
      *
      * The Catalyst runtime enables seeded program execution on non-hardware devices.
      * A random number generator instance is managed by the runtime to predictably
      * generate results for non-deterministic programs, such as those involving `Measure`
      * calls.
      * Devices implementing support for this feature do not need to use the provided
-     * PRNG instance as their sole source of random numbers, but it is expected that the
-     * the same instance state will predictable and reproducibly generate the same
-     * program results. It is also expected that the provided PRNG state is evolved
-     * sufficiently so that two device executions sharing the same instance do not produce
-     * identical results.
-     * The provided PRNG instance is not thread-locked, and devices wishing to share it
-     * across threads will need to provide their own thread-safety.
-     *
-     * @param gen The std::mt19937 PRNG object.
+     * PRNG instance as their sole source of randomness, but it is expected that the
+     * the same instance state will predictably and reproducibly generate the same
+     * program results (considering all random processes in the device).
+     * It is also expected that the provided PRNG state is evolved sufficiently so that two copies
+     * of a device provided with the same PRNG instance do not produce identical results when run
+     * in succession.
+     * Note that the provided PRNG instance is not thread-locked, and devices wishing to use it
+     * across internal threads will need to provide their own thread-safety.
+     *
+     * @param gen Pointer to a Catalyst-managed Mersenne Twister instance.
      */
     virtual void SetDevicePRNG([[maybe_unused]] std::mt19937 *gen){};
```

Comment:
Could you also mention the default behaviour of this method that does nothing?

Reply:
Default behaviour is to do nothing (i.e. no seeding).